modules/virt-running-real-time-checkup.adoc

```diff
@@ -0,0 +1,215 @@
++
+.Example service account, role, and rolebinding manifest file
+[%collapsible]
+====
+[source,yaml]
+----
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: realtime-checkup-sa
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: kiagnose-configmap-access
+rules:
+  - apiGroups: [ "" ]
+    resources: [ "configmaps" ]
+    verbs: [ "get", "update" ]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: kiagnose-configmap-access
+subjects:
+  - kind: ServiceAccount
+    name: realtime-checkup-sa
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: kiagnose-configmap-access
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: kubevirt-realtime-checker
+rules:
+  - apiGroups: [ "kubevirt.io" ]
+    resources: [ "virtualmachineinstances" ]
+    verbs: [ "create", "get", "delete" ]
+  - apiGroups: [ "subresources.kubevirt.io" ]
+    resources: [ "virtualmachineinstances/console" ]
+    verbs: [ "get" ]
+  - apiGroups: [ "" ]
+    resources: [ "configmaps" ]
+    verbs: [ "create", "delete" ]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: kubevirt-realtime-checker
+subjects:
+  - kind: ServiceAccount
+    name: realtime-checkup-sa
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: kubevirt-realtime-checker
+----
+====
+
+. Apply the `ServiceAccount`, `Role`, and `RoleBinding` manifest to the target namespace:
++
+[source,terminal]
+----
+$ oc apply -n <target_namespace> -f <realtime_sa_roles_rolebinding>.yaml
+----
+
+. Create a `ConfigMap` manifest file that contains the input parameters for the checkup:
++
+.Example input config map
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: realtime-checkup-config
+  labels:
+    kiagnose/checkup-type: kubevirt-realtime
+data:
+  spec.timeout: 1h10m <1>
+  spec.param.vmUnderTestContainerDiskImage: quay.io/kiagnose/kubevirt-realtime-checkup-vm:v0.2.0 <2>
+  spec.param.oslatDuration: 1h <3>
+----
+<1> The time, in hours and minutes, before the checkup fails.
+<2> The container disk image for the VM under test.
+<3> The duration for which the OS level thread latency test program runs.
+
+. Apply the `ConfigMap` manifest in the target namespace:
++
+[source,terminal]
+----
+$ oc apply -n <target_namespace> -f <real-time_config_map>.yaml
+----
+
+. Create a `Job` manifest to run the checkup:
++
+.Example job manifest
+[source,yaml,subs="attributes+"]
+----
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: realtime-checkup
+  labels:
+    kiagnose/checkup-type: kubevirt-realtime
+spec:
+  backoffLimit: 0
+  template:
+    spec:
+      serviceAccountName: realtime-checkup-sa
+      restartPolicy: Never
+      containers:
+        - name: realtime-checkup
+          image: registry.redhat.io/container-native-virtualization/kubevirt-realtime-checkup-rhel9:v{product-version}.0
```

Comment:
This how it looks in the preview:
```
vm-network-latency-checkup-rhel9:vBranch Build.0
```
(I'm talking specifically baout the `Branch Build` part).

Reply:
The `{product-version}` attribute will resolve to `4.16` in the live docs.